python/cudf/cudf/_libxx/column.pyx

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import cython
+import rmm
+
+from libc.stdint cimport uintptr_t
+from libcpp.pair cimport pair
+from libcpp cimport bool
+
+from cudf._libxx.lib cimport *
+from cudf.core.buffer import Buffer
+from libc.stdlib cimport malloc, free
+
+from cudf.utils.dtypes import is_categorical_dtype
+
+
+np_to_cudf_types = {np.dtype('int8'): INT8,
+                    np.dtype('int16'): INT16,
+                    np.dtype('int32'): INT32,
+                    np.dtype('int64'): INT64,
+                    np.dtype('float32'): FLOAT32,
+                    np.dtype('float64'): FLOAT64,
+                    np.dtype("datetime64[D]"): TIMESTAMP_DAYS,
+                    np.dtype("datetime64[s]"): TIMESTAMP_SECONDS,
+                    np.dtype("datetime64[ms]"): TIMESTAMP_MILLISECONDS,
+                    np.dtype("datetime64[us]"): TIMESTAMP_MICROSECONDS,
+                    np.dtype("datetime64[ns]"): TIMESTAMP_NANOSECONDS,
+                    np.dtype("object"): STRING,
+                    np.dtype("bool"): BOOL8
+}
+
+cudf_to_np_types = {INT8: np.dtype('int8'),
+                    INT16: np.dtype('int16'),
+                    INT32: np.dtype('int32'),
+                    INT64: np.dtype('int64'),
+                    FLOAT32: np.dtype('float32'),
+                    FLOAT64: np.dtype('float64'),
+                    TIMESTAMP_DAYS: np.dtype("datetime64[D]"),
+                    TIMESTAMP_SECONDS: np.dtype("datetime64[s]"),
+                    TIMESTAMP_MILLISECONDS: np.dtype("datetime64[ms]"),
+                    TIMESTAMP_MICROSECONDS: np.dtype("datetime64[us]"),
+                    TIMESTAMP_NANOSECONDS: np.dtype("datetime64[ns]"),
+                    STRING: np.dtype("object"),
+                    BOOL8: np.dtype("bool")
+}
+
+@cython.auto_pickle(True)
+cdef class Column:
+
+    def __init__(self, data, size, dtype, mask=None, offset=0, children=(), name=None):
+        if not isinstance(data, Buffer):
+            raise TypeError("Expected a Buffer for data, got " + type(data).__name__)
+        if mask is not None and not isinstance(mask, Buffer):
+            raise TypeError("Expected a Buffer for mask, got " + type(mask).__name__)
+        self.data = data
+        self.size = size
+        self.dtype = dtype
+        self.mask = mask
+        self.offset = offset
+        self.children = children
+        self.name = name
+
+    @property
+    def null_count(self):
+        return self.null_count()
+
+    cdef size_type null_count(self) except? 0:
+        return self.view().null_count()
+
+    cdef mutable_column_view mutable_view(self) except *:
+        if is_categorical_dtype(self.dtype):
+            data_dtype = self.codes.dtype
```

Comment:
```suggestion
            data_dtype = self.cat.codes.dtype
```

Reply:
This error may be elsewhere, need to scan again

Reply:
CategoricalColumns have a `codes` property, do we still want to go through the categorical accessor?